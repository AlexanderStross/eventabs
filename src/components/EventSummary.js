import React from 'react'
import PropTypes from 'prop-types'
import {
  MDBCol,
  MDBCard,
  MDBCardBody,
  MDBCardTitle,
  MDBCardText,
  MDBCardLink,
  MDBCardImage,
  MDBBtn,
  MDBRipple
} from 'mdb-react-ui-kit';
import moment from 'moment'


const EventSummary = props => (<MDBCol md={4}>
  <MDBCard style={{
      maxWidth: '22rem'
    }} className="mb-4 event-card">
    <MDBRipple rippleColor='light' rippleTag='div' className='bg-image hover-overlay'>
      <MDBCardImage src={props.event.image_url} fluid="fluid" alt='...'/>
      <a href={`/events/${props.event.id}`}>
        <div className='mask' style={{
            backgroundColor: 'rgba(251, 251, 251, 0.15)'
          }}></div>
      </a>
    </MDBRipple>
    <MDBCardBody>
      <MDBCardLink href={`/events/${props.event.id}`} className="event-card-link" style={{
          textDecoration: 'none',
          color: 'black'
        }}>
        <MDBCardTitle>{props.event.title}</MDBCardTitle>
      </MDBCardLink>
      <MDBCardText>
        <span className="event-datetime"> {moment(props.event.start_datetime).format('ddd, MMM DD, YYYY hh:mm A z')}</span>
      </MDBCardText>
      <MDBCardText>
        <span className="event-location">{props.event.location}</span>
      </MDBCardText>
      <MDBBtn href={`/events/${props.event.id}`}>View Event</MDBBtn>
    </MDBCardBody>
  </MDBCard>
</MDBCol>)

Event.propTypes = {
  event: PropTypes.shape({
    title: PropTypes.string.isRequired,
    start_datetime: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired
  })
}

export default EventSummary
