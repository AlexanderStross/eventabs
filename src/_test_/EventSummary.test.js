import React from "react";
import { render } from "@testing-library/react";
import EventSummary from "../components/EventSummary";
import renderer from "react-test-renderer";

it("renders event title", () => {
  const { getByText } = render(
    <EventSummary event={{ title: "My Test Event" }} />
  );
  const title = getByText("My Test Event");
  expect(title).toBeInTheDocument();
});

it("renders event start_datetime", () => {
  const { getByText } = render(
    <EventSummary
      event={{ start_datetime: new Date("10 September 2020, 10:00 am") }}
    />
  );
  const start_datetime = getByText("Thu, Sep 10, 2020 10:00 AM");
  expect(start_datetime).toBeInTheDocument();
});

it("renders event image", () => {
  const { getByRole } = render(
    <EventSummary event={{ image_url: "event.png" }} />
  );
  const image = getByRole("img", { src: "event.png" });
  expect(image).toBeInTheDocument();
});

it("should render a prettier event summary card", () => {
  const tree = renderer
    .create(
      <EventSummary
        event={{
          id: 1,
          title: "My Test Event",
          start_datetime: new Date("10 September 2020, 10:00 am"),
          image_url: "event.png",
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchInlineSnapshot(`
    <div
      className="col-md-4"
    >
      <div
        className="card mb-4 event-card"
        style={
          Object {
            "maxWidth": "22rem",
          }
        }
      >
        <div
          className="ripple ripple-surface bg-image hover-overlay"
          onClick={[Function]}
        >
          <img
            alt="..."
            className="img-fluid"
            src="event.png"
          />
          <a
            href="/events/1"
          >
            <div
              className="mask"
              style={
                Object {
                  "backgroundColor": "rgba(251, 251, 251, 0.15)",
                }
              }
            />
          </a>
        </div>
        <div
          className="card-body"
        >
          <a
            className="card-link event-card-link"
            href="/events/1"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <h5
              className="card-title"
            >
              My Test Event
            </h5>
          </a>
          <p
            className="card-text"
          >
            <span
              className="event-datetime"
            >
               
              Thu, Sep 10, 2020 10:00 AM 
            </span>
          </p>
          <p
            className="card-text"
          >
            <span
              className="event-location"
            />
          </p>
          <a
            className="ripple ripple-surface btn btn-primary"
            href="/events/1"
            onClick={[Function]}
            role="button"
          >
            View Event
          </a>
        </div>
      </div>
    </div>
  `);
});
