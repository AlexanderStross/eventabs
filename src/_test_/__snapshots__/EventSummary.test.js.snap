// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an event summary card 1`] = `
<div
  className="col-md-4"
>
  <div
    className="card mb-4 event-card"
    style={
      Object {
        "maxWidth": "22rem",
      }
    }
  >
    <div
      className="ripple ripple-surface bg-image hover-overlay"
      onClick={[Function]}
    >
      <img
        alt="..."
        className="img-fluid"
        src="event.png"
      />
      <a
        href="/events/1"
      >
        <div
          className="mask"
          style={
            Object {
              "backgroundColor": "rgba(251, 251, 251, 0.15)",
            }
          }
        />
      </a>
    </div>
    <div
      className="card-body"
    >
      <a
        className="card-link event-card-link"
        href="/events/1"
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <h5
          className="card-title"
        >
          My Test Event
        </h5>
      </a>
      <p
        className="card-text"
      >
        <span
          className="event-datetime"
        >
           
          Thu, Sep 10, 2020 10:00 AM 
        </span>
      </p>
      <p
        className="card-text"
      >
        <span
          className="event-location"
        />
      </p>
      <a
        className="ripple ripple-surface btn btn-primary"
        href="/events/1"
        onClick={[Function]}
        role="button"
      >
        View Event
      </a>
    </div>
  </div>
</div>
`;
